lib/vsc/filesystem/oceanstor.py

```diff
@@ -685,34 +818,30 @@ def make_fileset(self, new_fileset_path, fileset_name=None, parent_fileset_name=
 
         # Identify local mounted filesystem
-        local_fs = self.what_filesystem(dt_parentdir)
+        ostor_fs_id, ostor_dtree_id, ostor_mount, ostor_parentdir = self._identify_local_path(dt_parentdir)
 
         # Check type of OceanStor object mounted in this path
-        oceanstor_id = local_fs[self.localfilesystemnaming.index('oceanstor')]
-        oceanstor_fs_id, oceanstor_dtree_id = oceanstor_id.split('@', 1)
-        if int(oceanstor_dtree_id) == 0:
-            # This NFS mount contains a filesystem
-            oceanstor_fs_name = next(iter(self.select_filesystems(oceanstor_fs_id, byid=True)))
-            self.log.debug("NFS mount '%s' contains OceanStor filesystem '%s'", oceanstor_fs_name)
+        if int(ostor_dtree_id) == 0:
+            # this NFS mount contains a filesystem
+            ostor_fs_name = next(iter(self.select_filesystems(ostor_fs_id, byid=True)))
+            self.log.debug("NFS mount '%s' contains OceanStor filesystem '%s'", dt_fullpath, ostor_fs_name)
         else:
-            # This NFS mount contains a dtree fileset
-            errmsg = "NFS mount '%s' is already a dtree fileset (%s). Nested dtrees are not allowed in OceanStor."
-            self.log.raiseException(errmsg % (dt_fullpath, oceanstor_id), OceanStorOperationError)
+            # this NFS mount contains a dtree fileset
+            errmsg = "NFS mount '%s' is already a dtree fileset (%s@%s). Nested dtrees are not allowed in OceanStor."
+            self.log.raiseException(errmsg % (dt_fullpath, ostor_fs_id, ostor_dtree_id), OceanStorOperationError)
 
-        # Relative path of parent directory holding the fileset
-        local_mount = local_fs[self.localfilesystemnaming.index('mountpoint')]
-        dt_parentdir_relative = os.path.relpath(dt_parentdir, start=local_mount)
-        if dt_parentdir_relative.startswith('..'):
-            errmsg = "Parent directory of new fileset cannot be outside of filesystem boundaries. %s got: '%s'"
-            self.log.raiseException(errmsg % (fileset_name, dt_parentdir_relative), OceanStorOperationError)
-        elif dt_parentdir_relative == '.':
-            dt_parentdir_relative = ''
+        # Send API request to create the new dtree fileset
+        dbgmsg = "Sending request for new dtree fileset '%s' in OceanStor filesystem '%s' with parent directory '%s'"
+        self.log.debug(dbgmsg, fileset_name, ostor_fs_name, ostor_parentdir)
 
-        oceanstor_parentdir = '/' + dt_parentdir_relative
-        self.log.debug("Creating new dtree fileset with parent directory: '%s'", oceanstor_parentdir)
+        self.make_fileset_api(fileset_name, ostor_fs_name, parent_dir=ostor_parentdir)
 
-        # Send API request to create the new dtree fileset
-        self.make_fileset_api(fileset_name, oceanstor_fs_name, parent_dir=oceanstor_parentdir)
+        # wait for NFS lookup cache to expire to be able to access new fileset
+        time.sleep(NFS_LOOKUP_CACHE_TIME)
```

Comment:
on the VM that will run this, we don't really care about speed so we might as well disable the lookup cache.